src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke.command;
+
+import duke.Duke;
+import duke.data.DukeData;
+import duke.data.TaskList;
+import duke.task.*;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Deals with making sense of the user command,
+ * and simulates the supposed interaction from the command.
+ */
+public class Parser {
+    private Ui _ui;
+    private Duke _duke;
+    private DukeData _data;
+    private Scanner sc;
+
+    /**
+     * Creates a new Parser object with the given Duke program.
+     * @param parseDuke the Duke Program that will be handles by the Parser
+     */
+    public Parser(Duke parseDuke) {
+        this._duke = parseDuke;
+        this._data = parseDuke.getData();
+        this._ui = new Ui();
+        this.sc = new Scanner(System.in);
+        System.out.println(_ui.showIntro());
+    }
+
+    /**
+     * The parse method handles all input and output from the Scanner,
+     * as well as their corresponding interactions through the Duke program.
+     */
+    public void parse() {
+        TaskList tasks = this._data.load();
+        while (sc.hasNext()) {
+            String cmdWord = sc.next(); // extract first only
+            String cmdLine = sc.nextLine(); // extract the rest of the line
+
```

Comment:
I think this is a great way of parsing the line, at least much more efficient than my method of splitting the string